File: code/cpp/ch20/practice/challenge_03_making_large_island.cpp
#include <vector>
#include <queue>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <iostream>
using namespace std;

int solve(vector<vector<int>>& grid) {
    // TODO: Replace this with your solution
    return 0;
}

int main() {
    int n;
    cin >> n;
    vector<vector<int>> grid(n, vector<int>(n));
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++)
            cin >> grid[i][j];
    cout << solve(grid) << endl;
    return 0;
}

File: code/java/ch20/practice/Challenge03MakingLargeIsland.java
package ch20.practice;

import java.util.*;

/**
 * Challenge 3: Making a Large Island
 * Chapter 20: Graphs II — Real Problems
 *
 * PROBLEM: Flip at most one 0 to 1. Return largest island size.
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Challenge03MakingLargeIsland {
    public static int solve(int[][] grid) {
        // TODO: Replace this with your solution
        return 0;
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        int[][] grid = new int[n][n];
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                grid[i][j] = scan.nextInt();
        System.out.println(solve(grid));
        scan.close();
    }
}
